lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -482,6 +484,126 @@ Function Nssm-Update-Dependencies
     }
 }
 
+Function Convert-HashToString
+{
+    <#
+    Convert a hash into a variable length string of KEY=VALUE seperated by spaces
+
+    Input:
+    {
+        key1: value1,
+        key2: value2
+    }
+
+    Output:
+    "key1=value1 key2=value2"
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [hashtable]$hash
+    )
+
+    return ($hash.GetEnumerator() | % { "$($_.Key)=$($_.Value)" }) -join " "
+}
+
+Function Convert-ArrayToHash
+{
+    <#
+    Converts an array with key=value elements into a hashtable
+
+    Input:
+    ["key1=value1", "key2=value2"]
+
+    Output:
+    {
+        key1: value1,
+        key2: value2
+    }
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [array]$array
+    )
+
+    $hash = @{}
+
+    foreach ($element in $array)
+    {
+        $hash = $hash + (ConvertFrom-StringData $element)
+    }
+    return $hash
+}
+
+Function Nssm-Update-Extra-Environment
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$name,
+        [Parameter(Mandatory=$true)]
+        [hashtable]$env_vars
+    )
+
+    $env_vars_string = Convert-HashToString $env_vars
+
+    $cmd = "get ""$name"" AppEnvironmentExtra"
+    $results = Nssm-Invoke $cmd
+
+    if ($LastExitCode -ne 0)
+    {
+        $result.nssm_error_cmd = $cmd
+        $result.nssm_error_log = "$results"
+        Throw "Error updating environment variables for service ""$name"""
```

Comment:
I would change the error message to be more in line with what you are trying to do (get current env vars).